### ISSUE TITLE
Upgrade Apache Lucene library to 9.9.x for jena-text

### ISSUE THREAD

**Issue by afs** [apache/jena#2148]
### Change

This issue is to track dependabot PR #2137.

### Are you interested in contributing a pull request for this task?

Yes

**Comment by afs** [comment]
9.7.0 -> 9.8.0
https://lucene.apache.org/core/9_8_0/changes/Changes.html

9.8.0 -> 9.9.0
https://lucene.apache.org/core/9_9_0/changes/Changes.html

**Comment by afs** [comment]
Note:
An issue with 9.8.0 was that 21-EA could cause problems. Java21 final has been available for several months now.
https://github.com/apache/jena/issues/2045

With 9.9.1, the build, run with Java21  outputs:
```
INFO: Using MemorySegmentIndexInput with Java 21; 
     to disable start with -Dorg.apache.lucene.store.MMapDirectory.enableMemorySegments=false
```

Looks good for the update.

**Comment by afs** [comment]
Closed by #2137.
